app/controllers/listings_controller.rb

```diff
@@ -101,7 +101,13 @@ def numbers_str_to_array(str)
   # Used to show multiple listings in one bubble
   def listing_bubble_multiple
     ids = numbers_str_to_array(params[:ids])
-    @listings = Listing.visible_to(@current_user, @current_community, ids).order("id DESC")
+
+    if @current_user || !@current_community.private?
```

Comment:
This could be extracted to own function. Now `@listings` is initialized in two places. Something like:

```ruby
@listings = listings_for_community(@current_community)
```